main.go

```diff
@@ -3,64 +3,37 @@ package main
 import (
 	"context"
 	"net/http"
-	"os"
-	"strconv"
 	"time"
 
 	"github.com/NickGowdy/deveui-cli/client"
 	"github.com/NickGowdy/deveui-cli/processor"
-	"github.com/joho/godotenv"
+	"github.com/kelseyhightower/envconfig"
 )
 
-/*
-deveui-cli is a Go CLI program.
-It is used for concurrently generating unique 16-character (hex) identifier called a DevEUI.
-These are generated by the program and registered via an external (LoRaWAN) API.
-
-Usage:
-
-	go run main.go (locally)
-
-	go run deveui-cli (docker)
+type config struct {
+	BaseURL               string        `envconfig:"BASE_URL" default:"http://europe-west1-machinemax-dev-d524.cloudfunctions.net"`
+	MaxConcurrent         int           `envconfig:"MAX_CONCURRENT" default:"10"`
+	CodeRegistrationLimit int           `envconfig:"CODE_REGISTRATION_LIMIT" default:"100"`
+	Timeout               time.Duration `envconfig:"TIMEOUT" default:"30s"`
+}
 
-Once this program starts, it will listen to syscall.SIGTERM, syscall.SIGINT via a channel.
-This is to handle any unexpected terminations of the program and to resume processing DevEUIs.
-*/
 func main() {
-	if err := godotenv.Load(".env"); err != nil {
```

Comment:
I have a couple of issues with this. Having a .env file is fairly common, however if the file exists but a value is missing then this won't detect it. For example, if BASE_URL is missing, then os.Getenv will return with an empty string. If you wanted to handle that case then you'd need to do a check and set a default value. Another issue is the type conversions below, which is cumbersome and a lot of code. On the other hand, envconfig is very flexible, you can tell which env vars are required or optional, you can set default values and type conversion is dealt with too, See how much readable the code becomes with it.